src/redux/books/bookSlice.js

```diff
@@ -1,26 +1,39 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const dataUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/McvW73kTCQbVPzcD7S2A/books';
+
+export const getBooks = createAsyncThunk('book/getBooks', () => axios.get(dataUrl)
+  .then((res) => res.data)
+  .catch((err) => console.log(err)));
+
+export const postData = createAsyncThunk('book/postData', async (bookDetail) => fetch(' https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/McvW73kTCQbVPzcD7S2A/books ', {
+  method: 'POST',
+  body: JSON.stringify({
+    item_id: bookDetail.id,
+    title: bookDetail.title,
+    author: 'suzan collins',
+    category: bookDetail.category,
+  }),
+  headers: {
+    'Content-type': 'application/json',
+  },
+}));
+
+export const deleteData = createAsyncThunk('book/deleteData', (itemid) => fetch(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/McvW73kTCQbVPzcD7S2A/books/${itemid}`, {
```

Comment:
- Please update this function. Make it return a promise. This will allow you to display useful message as the item is getting deleted.